attestation/sample_vbs_protected_key_att.cpp

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+
+/**
+ * WARNING: Information regarding VBS-Protected Keys relates to prerelease product that may be substantially modified before it's commercially released. 
+ * Microsoft makes no warranties, express or implied, with respect to the information provided here.
+ */
+
+/**
+ * @brief This sample provides the code implementation to perform VBS-Protected key attestation,
+ * and retrieve an attestation token from Microsoft Azure Attestation.
+ *
+ * @remark The following environment variables must be set before running the sample.
+ *
+ * - AZURE_TENANT_ID:     Tenant ID for the Azure account. Used for authenticated calls to the attestation service.
+ * - AZURE_CLIENT_ID:     The client ID to authenticate the request. Used for authenticated calls to the attestation service.
+ * - AZURE_CLIENT_SECRET: The client secret. Used for authenticated calls to the attestation service.
+ * - AZURE_MAA_URI:       Microsoft Azure Attestation provider's Attest URI (as shown in portal). Format is similar to "https://<ProviderName>.<Region>.attest.azure.net".
+ *
+ * In addition, a TPM attestation identity key named 'att_sample_aik' must be created. See README.md for instructions.
+ *
+ * Finally, a fixed relying party id and nonce are used in this sample. An application should obtain a per-session nonce from the relying party before making
+ * the call to the attestation service. TODOs in the code below mark the locations to be updated.
+ *
+ */
+
+#include "utils.h"
+#include "attest.h"
+#include <string>
+#include <vector>
+#include <iostream>
+
+#include <att_manager.h>
+#include <att_manager_logger.h>
+
+using namespace std;
+
+#define AIK_NAME L"att_sample_aik"
+
+int main()
+{
+    // Adjust log level to your desired level of output. 
+    att_set_log_level(att_log_level_none);
+    att_set_log_listener(sample_log_listener);
+
+    // TODO: Use relying party's id in the line below.
+    string rp_id{ "https://contoso.com" };
+    // TODO: Use relying party's per-session nonce below.
+    vector<uint8_t> rp_nonce{ 'R', 'E','P','L','A','C','E',' ','W','I','T','H', ' ','R','P', ' ','N','O','N','C','E' };
+
+    try
+    {
+        auto tpm_aik = load_tpm_key(AIK_NAME, true);
+        auto vbs_protected_key = create_vbs_protected_key(L"att_sample_vbs_key", false);
```

Comment:
I used "att_sample_vbs_key" to distinguish from the key in the tpm key attestation sample, but I wanted to not make the name such a mouthful ("att_sample_vbs_protected_key"). If I should change the name for correctness, let me know.